api/types/clustername.go

```diff
@@ -142,10 +158,14 @@ func (c *ClusterNameV2) CheckAndSetDefaults() error {
 		return trace.BadParameter("cluster name is required")
 	}
 
+	if c.Spec.ClusterID == "" {
+		c.Spec.ClusterID = uuid.New()
```

Comment:
Try to keep `uuid` out of api to reduce dependencies. A generated uuid isn't exactly a `default`, it's more of a server side implementation detail, so I think it'd be better to keep this in `lib/auth` or wherever the clustername is being created if possible.

Reply:
Yes, empty `ClusterID` should be an error.